### ISSUE TITLE
[google_maps_flutter] Evaluate `google_maps` package dependency on web

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#130942]
See https://github.com/flutter/flutter/issues/122713 for context.

The `google_maps` package is very heavily used in the web implementation of `google_maps_flutter`, but it's outside the policy we've now established for dependencies. E.g., it looks like it may have a bus factor of one person currently, which is definitely not ideal.

@ditman @a14n Any thoughts on what we could do here? Is there a plausible path for this becoming part of flutter/packages, for instance?

**Comment by ditman** [comment]
I've never had any trouble with @a14n's ownership of this package, he's been stellar supporting it (see the 0 open issues).

IMO there isn't much that needs to be improved in the (published) version of the package, it works great and it saved us a lot of time! It does exactly what it says in the can.

As for *where* the source of the package is hosted, anywhere other than https://github.com/googlemaps is probably not the *best* location 😛 I personally don't think that any open source package needs to be moved in order to increase its bus factor; that can be achieved by more people (us) getting more involved with its development and maintenance.

Something we could do, for example, is maybe figure out a way to devote resources (code? infra?) to add tests of the generated output to ensure that "all the examples compile and run and do what's expected of them", for example, then keep building from there.

The package predates Flutter by about 4 years (!!), so if @a14n wants to keep it in his repo, I understand (and I'd rather find ways to contribute there). What I really, really (really) don't want is Alexandre to think that we're unhappy with how he's dealing with the package, or that we want to steal it from him.

**Comment by stuartmorgan** [comment]
> I've never had any trouble with @a14n's ownership of this package, he's been stellar supporting it (see the 0 open issues).

Yes sorry, I should have been more explicit about the context here; this is definitely not a judgment on the specific package or @a14n. E.g., https://github.com/flutter/flutter/issues/130915 is a very similar issue about other packages that are owned by other people on the Flutter team but under the Google org, not by the Flutter team itself. I'm looking at every dependency, not just this one :)

> The package predates Flutter by about 4 years (!!), so if @a14n wants to keep it in his repo, I understand (and I'd rather find ways to contribute there). What I really, really (really) don't want is Alexandre to think that we're unhappy with how he's dealing with the package, or that we want to steal it from him.

Absolutely, and if the outcome is that we don't change anything here for now that's reasonable; I just want to make sure we've at least discussed options for each one rather than having the dependencies that predate the explicit policy just continue unexamined. For all the packages that are currently owned by individual Flutter team members, bringing them into flutter/packages is something that I think is worth at least discussing as a potential option, but yes, that is 100% a decision that is up to @a14n. (And I absolutely wasn't suggesting a fork into flutter/packages, just the potential for a move if—and only if—everyone involved thought that would be a good outcome.)

**Comment by a14n** [comment]
Sorry for the late response. I don't have a strong opinion here. The google_maps package is not really active (the evolution are driven by user requests for new features available on the js api side). The current code is old and a little crappy. I hope to find time in future to rewrite the code and use the upcoming extension types. At this point it could be a good time to move the code where you suggested.

**Comment by alod2019** [comment]
Please take a look at https://github.com/flutter/flutter/issues/147492
